app/services/delivery_partners/search.rb

```diff
@@ -0,0 +1,19 @@
+module DeliveryPartners
+  class Search
+    def initialize(query_string = nil)
+      @scope = DeliveryPartner
+
+      @query_string = query_string
+    end
+
+    def search
+      query = if @query_string.blank?
+                @scope.all
+              else
+                @scope.where("name ILIKE ?", "%#{@query_string}%")
+              end
+
+      query.order(name: 'asc')
+    end
+  end
+end
```

Comment:
Does this work? It simplifies the code by removing an unnecessary instance variable.

```suggestion
module DeliveryPartners
  class Search
    def initialize(query_string = nil)
      @query_string = query_string
    end

    def search
      scope = DeliveryPartner.all

      if @query_string.present?
        scope = scope.where("name ILIKE ?", "%#{@query_string}%")
      end

      scope.order(name: :asc)
    end
  end
end
```